src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -0,0 +1,381 @@
+use crate::{
+    buffers::Acker,
+    event::{Event, Metric},
+    region::RegionOrEndpoint,
+    sinks::util::{
+        retries::{FixedRetryPolicy, RetryLogic},
+        BatchServiceSink, Partition, SinkExt,
+    },
+    topology::config::{DataType, SinkConfig},
+};
+use futures::{Future, Poll};
+use rusoto_cloudwatch::{
+    CloudWatch, CloudWatchClient, MetricDatum, PutMetricDataError, PutMetricDataInput,
+};
+use rusoto_core::{Region, RusotoFuture};
+use serde::{Deserialize, Serialize};
+use std::{convert::TryInto, time::Duration};
+use tower::{Service, ServiceBuilder};
+use tracing::field;
+use tracing_futures::{Instrument, Instrumented};
+
+#[derive(Clone)]
+pub struct CloudWatchMetricsService {
+    client: CloudWatchClient,
+    config: CloudWatchMetricsSinkConfig,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct CloudWatchMetricsSinkConfig {
+    pub namespace: String,
+    #[serde(flatten)]
+    pub region: RegionOrEndpoint,
+    pub batch_size: Option<usize>,
+    pub batch_timeout: Option<u64>,
+
+    // Tower Request based configuration
+    pub request_in_flight_limit: Option<usize>,
+    pub request_timeout_secs: Option<u64>,
+    pub request_rate_limit_duration_secs: Option<u64>,
+    pub request_rate_limit_num: Option<u64>,
+    pub request_retry_attempts: Option<usize>,
+    pub request_retry_backoff_secs: Option<u64>,
+}
+
+#[typetag::serde(name = "aws_cloudwatch_metrics")]
+impl SinkConfig for CloudWatchMetricsSinkConfig {
+    fn build(&self, acker: Acker) -> Result<(super::RouterSink, super::Healthcheck), String> {
+        let sink = CloudWatchMetricsService::new(self.clone(), acker)?;
+        let healthcheck = CloudWatchMetricsService::healthcheck(self)?;
+        Ok((sink, healthcheck))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Metric
+    }
+}
+
+impl CloudWatchMetricsService {
+    pub fn new(
+        config: CloudWatchMetricsSinkConfig,
+        acker: Acker,
+    ) -> Result<super::RouterSink, String> {
+        let client = Self::create_client(config.region.clone().try_into()?)?;
+
+        let batch_size = config.batch_size.unwrap_or(5);
```

Comment:
I don't thick there is a formula for this. However we are not using bytes as batching unit, but Events count. Each event is put into `PutMetricData` as separate `MetricDatum`. The limit for `MetricDatum` count per call is 20, and I think this could be a reasonable default for batch_size.